src/dev/Web/EastSeat.ResourceIdea.Web/Components/Pages/WorkItems/AddWorkItem.razor.cs

```diff
@@ -0,0 +1,296 @@
+            }
+        }
+        catch (InvalidOperationException ex)
+        {
+            Console.Error.WriteLine($"InvalidOperationException in LoadEmployees: {ex}");
+        }
+        catch (HttpRequestException ex)
+        {
+            Console.Error.WriteLine($"Network error in LoadEmployees: {ex}");
+        }
+        catch (TimeoutException ex)
+        {
+            Console.Error.WriteLine($"Timeout error in LoadEmployees: {ex}");
+        }
+        catch (Exception ex)
+        {
+            Console.Error.WriteLine($"Unexpected error in LoadEmployees: {ex}");
+            throw; // Rethrow unexpected exceptions to be handled at a higher level
+        }
+    }
+
+    private async Task HandleValidSubmit()
+    {
+        // Check if any engagements exist
+        if (!HasAnyEngagements)
+        {
+            NotificationService.ShowErrorNotification("No engagements available. Please add an engagement first before creating a work item.");
+            return;
+        }
+
+        // Set the engagement ID from the selected value
+        if (Guid.TryParse(SelectedEngagementId, out var engagementGuid))
+        {
+            Command.EngagementId = EastSeat.ResourceIdea.Domain.Engagements.ValueObjects.EngagementId.Create(engagementGuid);
+        }
+        else
+        {
+            NotificationService.ShowErrorNotification("Please select a valid engagement.");
+            return;
+        }
+
+        // Set the assigned employee ID if selected
+        if (!string.IsNullOrWhiteSpace(SelectedAssignedToId) && Guid.TryParse(SelectedAssignedToId, out var assignedToGuid))
+        {
+            Command.AssignedToId = EmployeeId.Create(assignedToGuid);
+        }
+        else
+        {
+            Command.AssignedToId = null; // Optional assignment
+        }
+
+        Command.TenantId = ResourceIdeaRequestContext.Tenant;
+        
+        ValidationResponse commandValidation = Command.Validate();
+        if (!commandValidation.IsValid && commandValidation.ValidationFailureMessages.Any())
+        {
+            // TODO: Log validation failure.
+            NotificationService.ShowErrorNotification("Validation failed. Please check the form and try again.");
+            return;
+        }
+
+        try
+        {
+            var response = await Mediator.Send(Command);
+            if (response.IsSuccess)
+            {
+                NotificationService.ShowSuccessNotification("Work item created successfully!");
+                NavigateBack();
+            }
+            else
+            {
+                NotificationService.ShowErrorNotification("Failed to create work item. Please try again.");
+            }
+        }
+        catch (Exception ex)
+        {
+            NotificationService.ShowErrorNotification("An unexpected error occurred. Please try again.");
+            Console.Error.WriteLine($"Error creating work item: {ex}");
```

Comment:
I am going to setup logging using Azure Application Insights, don't use `Console.Error.WriteLine`.